loader/icd_dispatch.c

```diff
@@ -18,9 +18,52 @@
 
 #include "icd_dispatch.h"
 #include "icd.h"
+#include "cl_icdl.h"
+#include "icd_version.h"
 #include <stdlib.h>
 #include <string.h>
 
+static cl_int
+clGetICDLoaderInfoOCLICD(
+    cl_icdl_info param_name,
+    size_t       param_value_size,
+    void *       param_value,
+    size_t *     param_value_size_ret)
+{
+    static const char  cl_icdl_OCL_VERSION[] = CL_ICDL_OCL_VERSION_STRING;
+    static const char  cl_icdl_VERSION[] = OPENCL_ICD_LOADER_VERSION_STRING;
+    static const char  cl_icdl_NAME[] = CL_ICDL_NAME_STRING;
+    static const char  cl_icdl_VENDOR[] = CL_ICDL_VENDOR_STRING;
+    size_t             pvs;
+    void *             pv;
+
+#define KHR_ICD_CASE_PARAM_NAME(name)                                          \
+    case CL_ICDL_ ## name:                                                     \
+        pvs = sizeof(cl_icdl_ ## name);                                        \
```

Comment:
This will work as long as these are all string literals or arrays (which they are currently), but I'd still consider calling strlen instead to avoid possible problems in the future.

Or, maybe this is the reason for the local variables above?

Reply:
As said above, I don't feel strongly about the implementation. I didn't have a particular template to follow, so I studied a few implementations (including ocl-icd) and came up with this, which as you justly remarked works if data are stored in array like variables (integers would, for instance, need to be defined as `static const int a[1] = {CL_ICDL_XXX};`).

The only opinion I have against `strlen` based approach is that they cannot be generalized to other parameter types, and it means adding new options will add type specific variants of the `KHR_ICD_CASE_PARAM_NAME` macro.

Reply:
The case I'd be a little worried about is the `foo3` case here: https://godbolt.org/z/oaEK6jbxr

This case returns size == 8 (the size of the pointer) rather than the length of the string.

Perhaps this isn't worth worrying about though...

Reply:
I update the macro, the assembly is the same. Tell me if you prefer it this way.

Reply:
Thanks!  I'm honestly fine either way.  I like the new macro name `KHR_ICD_CASE_STRING_PARAM_NAME` now though, and I think this will help avoid potential problems in the future.  Let's go with this for now and we can always revisit if needed.